x/wasm/client/cli/tx.go

```diff
@@ -94,23 +94,48 @@ func parseStoreCodeArgs(args []string, cliCtx client.Context) (types.MsgStoreCod
 	}
 
 	var perm *types.AccessConfig
-	if onlyAddrStr := viper.GetString(flagInstantiateByAddress); onlyAddrStr != "" {
+	onlyAddrStr, err := flags.GetString(flagInstantiateByAddress)
```

Comment:
This seems much more complex than using viper.

I guess it differentiates between not set (error) and set to empty sting as two different cases we handled the same before? Or is there any reason `flag.GetString()` would return an error besides being unset, which we actually want to allow, right? Huh

Reply:
Viper can not be used as before. The previous call was using a static instance but this has changed in the sdk. There is a viper instance behind the scenes but `flags` is also populated from the `app.toml` file with defaults.
`viper.GetString` simply ignored the error.